### PR TITLE
Unique cowboy minor NPCs

### PR DESCRIPTION
**Pull request by SirPiecemaker**
#### Summary

Summary: Content "Gives rural NPCs unique dialogue"

#### Purpose of change

Currently, the cowboy NPCs you can ocasionally find have merely generic dialogue. This creates a couple variants with unique dialogue that can be spawned to make the encounters a little more interesting.

#### Testing

Done in game via spawning.

### CODE REVIEW

File: data/json/npcs/holdouts/Rural - Russell.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   {
+   "type": "talk_topic",
+   "id": "TALK_ANDERSON_BACKGROUND",
+   "dynamic_line": "I worked on a ranch not too far from here before it all went down.  I'd love to get back to what I did - taking care of horses, cows, all sorts of animals, making sure the fields stay safe and all.  But now? I dunno, man.  Just trying to survive, I suppose.  Hopefully, the world will get back together one day and I can put my skills to use doing what I like.",
+   "responses": [ { "text": "Let's hope it will.", "topic": "TALK_NONE" } ]
+   },
+   {
+   "type": "talk_topic",
+   "id": "TALK_ANDERSON_INTRODUCTION",
+   "dynamic_line": "Name's Russel.  I was just sort of wandering around, hoping to maybe find some decent gear.  Decided to rest up, heard someone walking, and, well, here we are.",
```
Comment (Leland) [inline data/json/npcs/holdouts/Rural - Russell.json:44]:
```suggestion
   "dynamic_line": "Name's Russell.  I was just sort of wandering around, hoping to maybe find some decent gear.  Decided to rest up, heard someone walking, and, well, here we are.",
```

**Comment by Leland** [comment]
Isn't this game's setting supposed to be New England? I'm down for rural-specific dialog, but I don't think anyone that works on a farm in, say, Connecticut uses phrases like "partner" or "well I'll be"

You could look up rural Vermont accents to find some sources for exaggerated local dialects. The late Fred Tuttle's videos might provide a source, e.g [this one](https://www.youtube.com/watch?v=6MwICs8gIh4). Probably stuff like run on sentences, starting sentences with "Oh," etc.

File: data/json/npcs/classes.json [hunk 2]
```diff
@@ -97,7 +97,7 @@
     "type": "npc_class",
     "id": "NC_COWBOY",
     "name": { "str": "Cowboy" },
-    "job_description": "Just looking for some wrongs to right.",
+    "job_description": "I'm looking to make an honest living using the skills I gained in the country.",
```
Comment (I-am-Erk) [inline data/json/npcs/classes.json:100]: This line doesn't actually show up anywhere afaik

File: data/json/npcs/classes.json [hunk 3]
```diff
@@ -106,7 +106,8 @@
       { "skill": "ALL", "level": { "mul": [ { "one_in": 3 }, { "sum": [ { "dice": [ 2, 2 ] }, { "rng": [ 0, -4 ] } ] } ] } },
       { "skill": "gun", "bonus": { "rng": [ 1, 3 ] } },
       { "skill": "pistol", "bonus": { "rng": [ 1, 3 ] } },
-      { "skill": "rifle", "bonus": { "rng": [ 0, 2 ] } }
+      { "skill": "rifle", "bonus": { "rng": [ 0, 2 ] } },
+	  { "skill": "survival", "bonus": { "rng": [ 0, 2 ] } }
```
Comment (I-am-Erk) [inline data/json/npcs/classes.json:110]: This is fine, but I wouldn't muck around too much with skills and things... This is one of the classes that can show up as a random npc so it has a wide impact.

That said it's reasonable for them to get a bit more chance of having survival.

Comment (SirPiecemaker) [reply data/json/npcs/classes.json:110]: My thinking. Someone who lived on a farm/ranch would likely have some knowledge, and it's not like this tiny boost makes them crazy. Mostly for flavour.

**Comment by SirPiecemaker** [comment]
> Length wise this is fine, yeah. The NPC's I was picturing would still have random names and be slightly more vague about their identity, to avoid a player getting a horde of Russells following them. In the near future I want to be able to have NPC's introduce themselves using their random names anyway.

Yeah... that's the issue I ran into in the past too. Randomly generated name is better, but it's just weird not to have them actually introduce themselves. I could try and work around it in dialogue, but it will seem stiff until the name usage is done.

**Comment by SirPiecemaker** [comment]
> Isn't this game's setting supposed to be New England? I'm down for rural-specific dialog, but I don't think anyone that works on a farm in, say, Connecticut uses phrases like "partner" or "well I'll be"
> 
> You could look up rural Vermont accents to find some sources for some humorous and exaggerated local dialects. The late Fred Tuttle's videos might provide a source, e.g [this one](https://www.youtube.com/watch?v=6MwICs8gIh4). Probably stuff like run on sentences, starting sentences with "Oh," etc.

I get what you mean, and yeah, partner is proably on the nose, but at the same time, you can't convey an *accent* via written dialogue, unless it has some VERY specific telltale words and/or slang. You also can't put in intonations, pauses in talking, or a lot of less-wordy expressions, such as chuckling and such. Not like vast majority of the players would now what a New England accent sounds like, so even if I managed to make it sound accurate, it would just sound odd to a lot of players.

And I mean, I use "I'll be" in conversation, and english isn't even my first language.

File: data/json/npcs/classes.json (hunk 2 above)
Comment (SirPiecemaker) [reply data/json/npcs/classes.json:100]: Not for players, but on the off chance someone wants to use some NPCs somewhere, they might haphazardly spawn a couple of these thinking they're lawful good or something. It doesn't really matter in the end, so eh.

**Comment by jbytheway** [comment]
Would it be helpful to have a tag like `<mypronoun>` that is replaced by the NPC's name?  That would be trivial to add; just a few lines of code in [this function](https://github.com/CleverRaven/Cataclysm-DDA/blob/bd06e4d45857305302eb3de128f7385e7fd8c311/src/npctalk.cpp#L1253).  I'm rather surprised it's not there already...

**Comment by Leland** [comment]
>  I use "I'll be" in conversation

The exclamation "Well I'll be" – short for "Well I'll be damned" – is very southern; as used here:

https://github.com/CleverRaven/Cataclysm-DDA/blob/237bfa22a00dbee0ab7711fee53109f87523bf03/data/json/npcs/holdouts/Rural%20-%20Russell.json#L24

> Not like vast majority of the players would now what a New England accent sounds like, so even if I managed to make it sound accurate, it would just sound odd to a lot of players

Correctly conveying the game's regional accent would come across oddly? There are a lot of folks that live in that area – some 14 million people, myself included! Using some local reference material would be awesome!

Anyway, all I'm getting at is that there are ways to produce rural New England "hillbilly"/farm hand dialect that don't lean on Texan cowboy idioms. Though "accents" can't be conveyed in written form, you can evoke a rural dialect using techniques such as foreshortened words, sentence length, and misspelled words (e.g "summin" instead of "something").

**Comment by SirPiecemaker** [comment]
> > I use "I'll be" in conversation
> 
> The exclamation "Well I'll be" – short for "Well I'll be damned" – is very southern; as used here:
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/237bfa22a00dbee0ab7711fee53109f87523bf03/data/json/npcs/holdouts/Rural%20-%20Russell.json#L24
> 
> > Not like vast majority of the players would now what a New England accent sounds like, so even if I managed to make it sound accurate, it would just sound odd to a lot of players
> 
> Correctly conveying the game's regional accent would come across oddly? There are a lot of folks that live in that area – some 14 million people, myself included! Using some local reference material would be awesome!
> 
> Anyway, all I'm getting at is that there are ways to produce rural New England "hillbilly"/farm hand dialect that don't lean on Texan cowboy idioms. Though "accents" can't be conveyed in written form, you can evoke a rural dialect using techniques such as foreshortened words, sentence length, and misspelled words (e.g "summin" instead of "something").

So I looked into it a bit more, and turns out, New England accent is adjecant to a Boston one. That, I can do. Mostly take my inspiration from the most Boston sounding man I ever heard.

Reference: https://youtu.be/mMiBDW-6WK0

**Comment by SirPiecemaker** [comment]
Hey Leland - try and have a gander at this. I don't think I can lay an accent off any harder via text, but tell me what you think, since it's your native.

**Pull request merged**